Firebase/InstanceID/FIRInstanceID.m

```diff
@@ -1127,9 +1127,22 @@ - (void)observeFirebaseMessagingTokenChanges {
 - (void)messagingTokenDidChangeNotificationReceived:(NSNotification *)notification {
   NSString *tokenUpdatedFromMessaging = notification.object;
   if (!tokenUpdatedFromMessaging || [tokenUpdatedFromMessaging isKindOfClass:[NSString class]]) {
-    self.defaultFCMToken = tokenUpdatedFromMessaging;
-    [self.tokenManager saveDefaultToken:tokenUpdatedFromMessaging
-                            withOptions:[self defaultTokenOptions]];
+    // Check the token from storage along with local value.
+    FIRInstanceIDTokenInfo *cachedTokenInfo =
+        [self.tokenManager cachedTokenInfoWithAuthorizedEntity:self.fcmSenderID
+                                                         scope:kFIRInstanceIDDefaultTokenScope];
+    NSString *cachedToken = cachedTokenInfo.token;
+
+    if (self.defaultFCMToken.length != tokenUpdatedFromMessaging.length ||
```

Comment:
Why are the length checks needed? Aren't two `isEqualToString` checks sufficient?

Reply:
In case one of the string is null, and a [null isEqualtoString: null] in valid.